src/blocks/opentable/edit.js

```diff
@@ -0,0 +1,184 @@
+/* global coblocksBlockData */
+
+/**
+ * External dependencies
+ */
+import { OpentableIcon as icon } from '@godaddy-wordpress/coblocks-icons';
+
+/**
+ * Internal dependencies
+ */
+import Controls from './controls';
+import OpenTable from './opentable';
+import InspectorControls from './inspector';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { compose, usePrevious } from '@wordpress/compose';
+import {
+	Placeholder,
+	Notice,
+	Button,
+	withNotices,
+	Icon,
+	FormTokenField,
+} from '@wordpress/components';
+
+const Edit = ( props ) => {
+	const [ ridField, setRidField ] = useState( [] );
+	const [ queryResults, setQueryResults ] = useState( [] );
+	const { className, attributes } = props;
+	const [ noResultsFound, setNoResultsFound ] = useState( false );
+
+	const prevIDs = usePrevious( attributes.restaurantIDs );
+
+	//searches the opentable reservation network for restaurants with the given name or ID
+	const searchRestaurants = ( token ) => {
+		fetch(
+			'https://www.opentable.com/widget/reservation/restaurant-search?pageSize=15' +
+				'&query=' +
+				encodeURIComponent( token )
+		)
+			.then( ( response ) => response.json() )
+			.then( ( json ) => {
+				const results = [];
+				for ( const item in json.items ) {
+					const r = json.items[ item ];
+					const name = `${ decodeURIComponent( r.name ) },  ${ decodeURIComponent( r.addressResponse.city ) }, ${ decodeURIComponent( r.addressResponse.country ) } (${ r.rid })`;
+					results.push( name );
+				}
+				setQueryResults( results );
+				if ( results.length === 0 ) {
+					setNoResultsFound( true );
+				} else {
+					setNoResultsFound( false );
+				}
+			} );
+	};
+	useEffect( () => {
+		if ( prevIDs !== attributes.restaurantIDs ) {
+			setRidField( attributes.restaurantIDs );
+		}
+	}, [ attributes.restaurantIDs ] );
+
+	useEffect( () => {
+		if ( attributes.language === '' && typeof coblocksBlockData.localeCode !== 'undefined' ) {
```

Comment:
In this conditional, if `coblocksBlockDatra.localCode` is undefined then the default switch action does not fire. Is this the intended behavior?
```
default:
	props.setAttributes( { language: 'en-US' } );
	break;
```

I was just thinking that you could create a function 
```javascript
const getLocaleCode = () => {
  if ( typeof coblocksBlockData.localeCode !== 'undefined' ) {
    return coblocksBlockData.localeCode;
  }
  return '';
};
```

Then you could always run the switch on mount against the custom function. If for some reason there is an undefined localeCode then it will still result in the default action.

Reply:
This is a good suggestion. The original intended behavior was to leave the language as `''` in `attributes`, which still renders the widget correctly, and displays this in the language selector to prompt the user to select a language: 
<img width="277" alt="image" src="https://user-images.githubusercontent.com/9936886/128067584-9f06b1a0-2b94-4bb4-8b3c-5c885bbd4a85.png">

The better solution is to always set a language if unset. I assume this is the safer route, so resolved in 897960c